src/gausskernel/storage/access/k2/session.cpp

```diff
@@ -106,27 +105,35 @@ boost::future<sh::Response<>> TxnManager::beginTxn() {
             if (status.is2xxOK()) {
                 K2LOG_D(log::k2pg, "Started new txn: {}", handle);
                 _txn = std::make_unique<sh::TxnHandle>(std::move(handle));
+            } else {
+                K2LOG_E(log::k2pg, "Unable to begin txn due to: {}", status);
             }
-            K2LOG_E(log::k2pg, "Unable to begin txn due to: {}", status);
             return sh::Response<>(std::move(status));
         });
     }
     K2LOG_D(log::k2pg, "Found existing txn");
     return sh::MakeResponse<>(std::move(status));
 }
 
-boost::future<sh::Response<>> TxnManager::endTxn(sh::dto::EndAction endAction) {
+boost::future<sh::Response<>> TxnManager::endTxn(sh::dto::EndAction endAction, bool ignoreNotExists) {
```

Comment:
There is no need to complicate the code with a boolean flag. We already return a status which can signal a particular condition. The caller can choose how to handle the given status.

As I mentioned in standup, in this particular scenario you should handle in the caller the 410Gone status and ignore it.

Reply:
Without the boolean flag, it prints log warning that txn not found.